azure-iot-device/tests/provisioning/pipeline/test_provisioning_pipeline.py

```diff
@@ -327,226 +312,129 @@ def test_op_fail(self, mocker, pipeline, arbitrary_exception):
 
 @pytest.mark.describe("ProvisioningPipeline - .register()")
 class TestSendRegister(object):
-    @pytest.mark.it("Request calls publish on provider")
-    def test_send_register_request_calls_publish_on_provider(
-        self, mocker, mock_provisioning_pipeline, input_security_client, mock_mqtt_transport
-    ):
-        mock_init_uuid = mocker.patch(
-            "azure.iot.device.common.pipeline.pipeline_stages_base.uuid.uuid4"
+    @pytest.mark.it("Runs a RegisterOperation on the pipeline")
+    def test_runs_op(self, pipeline, mocker):
+        cb = mocker.MagicMock()
+        pipeline.register(callback=cb)
+        assert pipeline._pipeline.run_op.call_count == 1
+        assert isinstance(
+            pipeline._pipeline.run_op.call_args[0][0], pipeline_ops_provisioning.RegisterOperation
```

Comment:
Also validate the contents of the operation: the registration id and request payload